### PR TITLE
Preliminary build support for macOS on aarch64 (M1)

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

Our current and future contributors can't build Armeria with their M1 MacBooks.

Modifications:

- Fixed the test cases that make false assumption that Brotli codec will
  be always available.
- Fixed `FileService` so that it works as expected even when Brotli
  codec is not available.
- Updated NPM to 8.5.2 for `docs-client`
- Updated `docs-client/package-lock.json`
- Fixed eslint and TypeScript compilation errors
  - Disabled soon-to-be-deprecated eslint rule.
- Fixed a Kotlin compiler warning in `DataClassDocServiceTest`
- Added `-Xskip-prerelease-check` option to Kotlin compiler to work
  around its complaints about the code compiled with a pre-release compiler.
- Misc:
  - Increased the timeout for some assertions, clients and services for
    less flakiness.

Result:

- A contributor can build Armeria on macOS with Apple Silicon (M1).
  - Known issues:
    - `:site` cannot be built yet. Need to upgrade a lot of
      dependencies to make it happen.
    - Parallel build sometimes results in a build failure related with
      duplicate classes in a JAR.
- (defect) `FileService` doesn't choose a `.br` file for generating
  decompressed content anymore when Brotli is not available.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -193,6 +194,15 @@ configure(projectsWithFlags('java')) {
     }
 }
 
+// Configure all Kotlin projects
+configure(projectsWithFlags('kotlin')) {
+    // Suppress the 'Class X is compiled by a pre-release version of Kotlin and
+    // cannot be loaded by this version of the compiler' error.
+    tasks.withType(KotlinCompile) { task ->
+        task.kotlinOptions.freeCompilerArgs += '-Xskip-prerelease-check'
```
Comment (ikhoon) [inline build.gradle:202]: Should we move the option to? https://github.com/line/armeria/blob/11c6233d556f44a1aaf2a309e15ec399d53d2144/gradle/scripts/lib/kotlin.gradle#L6

**Review by ikhoon — APPROVED** [review]
Thanks, @trustin!

File: build.gradle (hunk 1 above)
Comment (trustin) [reply build.gradle:202]: We can. Let me send out a PR there and merge it.

Comment (trustin) [reply build.gradle]: https://github.com/line/gradle-scripts/pull/116

File: core/src/test/java/com/linecorp/armeria/server/management/ManagementServiceTest.java [hunk 2]
```diff
@@ -84,7 +85,9 @@ void threadDumpWithJson() throws Exception {
 
     @Test
     void heapDump() throws InterruptedException {
-        final WebClient client = WebClient.of(server.httpUri());
+        final WebClient client = WebClient.builder(server.httpUri())
+                                          .responseTimeout(Duration.ofSeconds(45)) // Heap dump can take time.
```
Comment (minwoox) [inline core/src/test/java/com/linecorp/armeria/server/management/ManagementServiceTest.java:89]: It seems like the request timeout of the server is not increased?

**Review by minwoox — APPROVED** [review]
Thanks a lot! 👍

File: core/src/test/java/com/linecorp/armeria/server/management/ManagementServiceTest.java (hunk 2 above)
Comment (trustin) [reply core/src/test/java/com/linecorp/armeria/server/management/ManagementServiceTest.java:90]: Increased to 50 seconds

**Pull request merged**